suite/tests/client-interface/drreg-test.dll.c

```diff
@@ -313,10 +345,165 @@ event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *inst
+                                                   opnd_create_reg(DR_REG_XMM0)));
+        res = drreg_unreserve_register(drcontext, bb, inst, reg);
+        CHECK(res == DRREG_SUCCESS, "unreserve of xmm register should work");
+#endif
+    } else if (subtest == DRREG_TEST_19_C) {
+#ifdef __AVX__
+        dr_log(drcontext, DR_LOG_ALL, 1, "drreg test #19\n");
+        res = drreg_reserve_register_ex(drcontext, DRREG_SIMD_XMM_SPILL_CLASS, bb, inst,
+                                        &xmm0_allowed, &reg);
+        CHECK(reg == DR_REG_XMM0, "reserved non-allowed register error");
+        instrlist_meta_preinsert(bb, inst,
+                                 INSTR_CREATE_pcmpeqd(drcontext,
+                                                      opnd_create_reg(DR_REG_XMM0),
+                                                      opnd_create_reg(DR_REG_XMM0)));
+        res = drreg_unreserve_register(drcontext, bb, inst, reg);
+        CHECK(res == DRREG_SUCCESS, "unreserve of xmm register should work");
+#endif
+    } else if (subtest == DRREG_TEST_20_C) {
+#ifdef __AVX__
+        dr_log(drcontext, DR_LOG_ALL, 1, "drreg test #20\n");
+        res = drreg_reserve_register_ex(drcontext, DRREG_SIMD_XMM_SPILL_CLASS, bb, inst,
+                                        &xmm0_allowed, &reg);
+        CHECK(reg == DR_REG_XMM0, "reserved non-allowed register error");
+        instrlist_meta_preinsert(bb, inst,
+                                 INSTR_CREATE_pxor(drcontext,
+                                                   opnd_create_reg(DR_REG_XMM0),
+                                                   opnd_create_reg(DR_REG_XMM0)));
+        res = drreg_unreserve_register(drcontext, bb, inst, reg);
+        CHECK(res == DRREG_SUCCESS, "unreserve of xmm register should work");
+#endif
+    } else if (subtest == DRREG_TEST_21_C) {
+#ifdef __AVX__
+        dr_log(drcontext, DR_LOG_ALL, 1, "drreg test #21\n");
+        res = drreg_reserve_register_ex(drcontext, DRREG_SIMD_XMM_SPILL_CLASS, bb, inst,
+                                        &xmm0_allowed, &reg);
+        CHECK(reg == DR_REG_XMM0, "reserved non-allowed register error");
+        instrlist_meta_preinsert(bb, inst,
+                                 INSTR_CREATE_pcmpeqd(drcontext,
+                                                      opnd_create_reg(DR_REG_XMM0),
+                                                      opnd_create_reg(DR_REG_XMM0)));
+        res = drreg_unreserve_register(drcontext, bb, inst, reg);
+        CHECK(res == DRREG_SUCCESS, "unreserve of xmm register should work");
+#endif
+    } else if (subtest == DRREG_TEST_22_C) {
+#ifdef X86
+        dr_log(drcontext, DR_LOG_ALL, 1, "drreg test #22\n");
+        res = drreg_reserve_register_ex(drcontext, DRREG_SIMD_XMM_SPILL_CLASS, bb, inst,
+                                        NULL, &reg);
+        CHECK(reg == DR_REG_XMM7, "most efficient register not reserved");
```

Comment:
Add a comment on why DR_REG_XMM7 is the most efficient register in this test case